### PR TITLE
Add docker-compose to run UI and API together

### PR DESCRIPTION
**Pull request by schbetsy**
Closes https://github.com/cfpb/hmda-platform/issues/455

### CODE REVIEW

**Comment by schbetsy** [comment]
This is now ready for review. 

If you want to test on your machine, please follow the instructions I added to the README (and provide feedback to help me improve those instructions :) ). However, instead of using the `master` branch of `hmda-platform-ui`, use the `update-docker-nginx` branch, which is in [UI PR #175](https://github.com/cfpb/hmda-platform-ui/pull/175l)

**Comment by awolfe76** [comment]
I'll test this afternoon but wanted to ask if creating a `docker-compose-dev.yml` file to do a few extra things for dev purposes should be part of this PR or should we create another issue to handle that? Either way works for me. 

Things like adding a `volumes:` on the UI side to allow us to still watch for changes during development ([an example in grasshopper](https://github.com/cfpb/grasshopper/blob/master/docker-compose.yml#L38)).

**Comment by schbetsy** [comment]
I'd be happy to add a second `docker-compose` with more dev-friendly options. I'd vote we make that a separate issue for a future PR, just because I don't know much about what we want/how to do it, and I don't want to delay this PR.

**Comment by hkeeler** [comment]
@awolfe76, I agree that it'd be nice to get the `volumes` setup like grasshopper.  I don't think there's a need for a separate Compose file, let's just be added right into the setup that @schbetsy created.

The grasshopper setup has:

```yaml
  volumes:
    - ../grasshopper-ui/dist:/usr/src/app/dist/
```

For hmda-platform-ui, would it just be:

```yaml
  volumes:
    - ../hmda-platform-ui/dist:/usr/src/app/dist/
```

...or is that app structured differently?

I'm also fine with this being a followup PR.  It'd probably be easier for @wpears or @awolfe76 to make that change, and be sure it's working the way they want.

**Comment by awolfe76** [comment]
I just created #464 to update `docker-compose` for development.

File: docker-compose.yml [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
+# Assumes UI repo is cloned in parent directory
+version: '2'
+services:
+  api:
+    build: .
+  ui:
+    build: ../hmda-platform-ui
+    ports:
+      - "80:80"
+    depends_on:
```
Comment (hkeeler) [inline docker-compose.yml]: Does `api` actually resolve to something in the container?  I thought that only works with [`links`](https://docs.docker.com/compose/compose-file/#/links), and not [`depends_on`](https://docs.docker.com/compose/compose-file/#/depends-on)?

**Comment by awolfe76** [comment]
After much docker, -compose, -machine debugging I got this working.

👍  from me. Nice work @schbetsy!

File: docker-compose.yml (hunk 1 above)
Comment (schbetsy) [inline docker-compose.yml]: When I run `docker exec -ti hmdaplatform_ui_1 /bin/bash`, I can `curl http://api:8080/` and see this:
```
root@c540789ba433:/usr/src/app# curl http://api:8080/
{
  "status": "OK",
  "service": "hmda-api",
  "time": "2016-07-29T15:33:24.876Z",
  "host": "44e2e13fb69c"
}root@c540789ba433:/usr/src/app#
```

Also, I can see the correct api responses in the browser by visiting `/api`, `/api/institutions`, etc.

**Pull request merged**